packages/full-stack-tests/src/hierarchy-builder/grouping/LabelGrouping.test.ts

```diff
@@ -0,0 +1,360 @@
+          const childPartition4 = insertPhysicalPartition({ builder, codeValue: "B4", parentId: rootSubject.id, userLabel: labelGroupName2 });
+          const childPartition5 = insertPhysicalPartition({ builder, codeValue: "B5", parentId: rootSubject.id, userLabel: labelGroupName2 });
+          return { rootSubject, childSubject1, childSubject2, childPartition3, childPartition4, childPartition5 };
+        });
+
+        const customHierarchy: IHierarchyLevelDefinitionsFactory = {
+          async defineHierarchyLevel(parentNode) {
+            if (!parentNode) {
+              return [
+                {
+                  fullClassName: subjectClassName,
+                  query: {
+                    ecsql: `
+                              SELECT ${await selectClauseFactory.createSelectClause({
+                                ecClassId: { selector: `this.ECClassId` },
+                                ecInstanceId: { selector: `this.ECInstanceId` },
+                                nodeLabel: "root subject",
+                              })}
+                              FROM ${subjectClassName} AS this
+                              WHERE this.ECInstanceId = (${IModel.rootSubjectId})
+                            `,
+                  },
+                },
+              ];
+            } else if (HierarchyNode.isInstancesNode(parentNode) && parentNode.label === "root subject") {
+              return [
+                {
+                  fullClassName: subjectClassName,
+                  query: {
+                    ecsql: `
+                        SELECT ${await selectClauseFactory.createSelectClause({
+                          ecClassId: { selector: `this.ECClassId` },
+                          ecInstanceId: { selector: `this.ECInstanceId` },
+                          nodeLabel: { selector: `this.UserLabel` },
+                          groupByClass: true,
+                          groupByLabel: true,
+                        })}
+                        FROM ${subjectClassName} AS this
+                        WHERE this.Parent.Id = (${IModel.rootSubjectId})
+                      `,
+                  },
+                },
+                {
+                  fullClassName: physicalPartitionClassName,
+                  query: {
+                    ecsql: `
+                        SELECT ${await selectClauseFactory.createSelectClause({
+                          ecClassId: { selector: `this.ECClassId` },
+                          ecInstanceId: { selector: `this.ECInstanceId` },
+                          nodeLabel: { selector: `this.UserLabel` },
+                          groupByClass: true,
+                          groupByLabel: true,
+                        })}
+                        FROM ${physicalPartitionClassName} AS this
+                        WHERE this.Parent.Id = (${IModel.rootSubjectId})
+                      `,
+                  },
+                },
```

Comment:
I believe this would be re-written as

```ts
{
  fullClassName: `BisCore.InformationContentElement`,
  query: {
    ecsql: `
      SELECT ${await selectClauseFactory.createSelectClause({
        ecClassId: { selector: `this.ECClassId` },
        ecInstanceId: { selector: `this.ECInstanceId` },
        nodeLabel: { selector: `this.UserLabel` },
        groupByClass: true,
        groupByLabel: true,
      })}
      FROM (
        SELECT ECClassId, ECInstanceId, UserLabel, Parent
        FROM ${subjectClassName}
        UNION ALL
        SELECT ECClassId, ECInstanceId, UserLabel, Parent
        FROM ${physicalPartitionClassName}
      ) AS this
      WHERE this.Parent.Id = (${IModel.rootSubjectId})
    `,
  },
},
```

Reply:
Applied your suggestion.